app/src/main/java/nerd/tuxmobil/fahrplan/congress/schedule/FahrplanFragment.kt

```diff
@@ -0,0 +1,739 @@
+            requireActivity().runOnUiThread { reloadHighlights() }
+        }
+    }
+
+    override fun onAttach(context: Context) {
+        super.onAttach(context)
+        appRepository = AppRepository
+        alarmServices = AlarmServices.newInstance(context, appRepository)
+        navigationMenuEntriesGenerator = NavigationMenuEntriesGenerator(dayString = getString(R.string.day), todayString = getString(R.string.today))
+        onSessionClickListener = if (context is OnSessionClickListener) {
+            context
+        } else {
+            throw IllegalStateException("$context must implement OnSessionClickListener")
+        }
+    }
+
+    override fun onDetach() {
+        onSessionClickListener = null
+        super.onDetach()
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setHasOptionsMenu(true)
+        val context = requireContext()
+        roomTitleTypeFace = TypefaceFactory.getNewInstance(context).getTypeface(Font.Roboto.Light)
+        sessionViewDrawer = SessionViewDrawer(context, { sessionPadding })
+        errorMessageFactory = ErrorMessage.Factory(context)
+    }
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+        val layoutRootView = inflater.inflate(R.layout.schedule, container, false)
+        val verticalScrollView = layoutRootView.requireViewByIdCompat<NestedScrollView>(R.id.verticalScrollView)
+        verticalScrollView.setOnScrollChangeListener(
+            OnScrollChangeListener { _, _, _, _, _ -> preserveVerticalScrollPosition = true }
+        )
+        return layoutRootView
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        val context = view.context
+        displayDensityScale = resources.displayMetrics.density
+        val roomScroller = view.requireViewByIdCompat<HorizontalScrollView>(R.id.roomScroller)
+        val snapScroller = view.requireViewByIdCompat<HorizontalSnapScrollView>(R.id.horizScroller)
+        snapScroller.setChildScroller(roomScroller)
+        roomScroller.setOnTouchListener { _, _ -> true }
+        mDay = appRepository.readDisplayDayIndex()
+        inflater = context.getLayoutInflater()
+        val intent = requireActivity().intent
+        sessionId = intent.getStringExtra(BUNDLE_KEY_SESSION_ALARM_SESSION_ID)
+        if (sessionId != null) {
+            MyApp.LogDebug(LOG_TAG, "Open with sessionId '$sessionId'.")
+            mDay = intent.getIntExtra(BUNDLE_KEY_SESSION_ALARM_DAY_INDEX, mDay)
+            MyApp.LogDebug(LOG_TAG, "day $mDay")
+        }
+        if (MyApp.meta.numDays > 1) {
+            buildNavigationMenu()
+        }
+    }
+
+    private fun saveCurrentDay(day: Int) {
+        appRepository.updateDisplayDayIndex(day)
+    }
+
+    override fun onResume() {
+        Log.d(LOG_TAG, "FahrplanFragment#onResume")
```

Comment:
:+1: The usage of logging in this class is subject to be cleaned up. Currently, three different patterns are used. I like to leave this for a follow-up pull request because I don't want to change to much in the Kotlin conversion of the class to keep it comprehensible.

Reply:
fair enough